dkg/sync/server.go

```diff
@@ -170,26 +170,40 @@ func (s *Server) setShutdown(pID peer.ID) {
 	s.shutdown[pID] = struct{}{}
 }
 
-// setStep sets the peer's reported step.
-func (s *Server) setStep(pID peer.ID, step int) {
+// updateStep updates the peer's step from the reported value.
+// Returns error if the reported step is not the same or monotonically increased.
+func (s *Server) updateStep(pID peer.ID, step int) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	currentPeerStep, hasCurrentPeerStep := s.steps[pID]
+	if hasCurrentPeerStep {
+		if step < currentPeerStep {
+			return errors.New("peer reported step is behind the last known step", z.Int("peer_step", step), z.Int("last_step", currentPeerStep))
+		} else if step > currentPeerStep+1 {
+			return errors.New("peer reported step is ahead the last known step", z.Int("peer_step", step), z.Int("last_step", currentPeerStep))
+		}
+	} else if step < 0 || step > 1 {
+		return errors.New("peer reported abnormal initial step, expected 0 or 1", z.Int("peer_step", step))
+	}
```

Comment:
Instead of branching out, i'd suggest we "return early". That is the guideline that we follow across the charon codebase. 

Suggest this instead:
```suggestion
	if hasCurrentPeerStep && step < currentPeerStep {
		return errors.New("peer reported step is behind the last known step", z.Int("peer_step", step), z.Int("last_step", currentPeerStep))
	}
	
	if hasCurrentPeerStep && step > currentPeerStep+1 {
		return errors.New("peer reported step is ahead the last known step", z.Int("peer_step", step), z.Int("last_step", currentPeerStep))
	}

	if step < 0 || step > 1 {
		return errors.New("peer reported abnormal initial step, expected 0 or 1", z.Int("peer_step", step))
	}
```

Reply:
Thank you for the hint, accepted!